File: src/com/kedzie/vbox/api/IVRDEServer.java
package com.kedzie.vbox.api;

import java.util.HashMap;
import java.util.Map;

import android.os.Parcel;
import android.os.Parcelable;

import com.kedzie.vbox.api.jaxb.AuthType;
import com.kedzie.vbox.soap.Asyncronous;
import com.kedzie.vbox.soap.KSOAP;
import com.kedzie.vbox.soap.VBoxSvc;

public interface IVRDEServer extends IManagedObjectRef, Parcelable {
	public static final String BUNDLE = "vrde";
	static final ClassLoader LOADER = IVRDEServer.class.getClassLoader();
    
    public static final Parcelable.Creator<IVRDEServer> CREATOR = new Parcelable.Creator<IVRDEServer>() {
		public IVRDEServer createFromParcel(Parcel in) {
			VBoxSvc vmgr =  in.readParcelable(LOADER);
			String id = in.readString();
			Map<String, Object> cache = new HashMap<String, Object>();
			in.readMap(cache, LOADER);
			return (IVRDEServer) vmgr.getProxy(IVRDEServer.class, id, cache); 
		}
		public IVRDEServer[] newArray(int size) {  
			return new IVRDEServer[size]; 
		}
	};
	
	@KSOAP(cacheable=true) public boolean getEnabled();
	@Asyncronous public void setEnabled(@KSOAP("enabled") boolean enabled);

	@KSOAP(cacheable=true) public AuthType getAuthType();
	@Asyncronous public void setAuthType(@KSOAP("authType") AuthType authType);
	     
	@KSOAP(cacheable=true) public int getAuthTimeout();
	@Asyncronous public void setAuthTimeout(@KSOAP(type="unsignedInt", value="authTimeout") int authTimeout);
     
	@KSOAP(cacheable=true) public boolean getAllowMultiConnection();
	@Asyncronous public void setAllowMultiConnection(@KSOAP("allowMultiConnection") boolean allowMultiConnection);
	
	@KSOAP(cacheable=true) public boolean getReuseSingleConnection();
	@Asyncronous public void setReuseSingleConnection(@KSOAP("reuseSingleConnection") boolean reuseSingleConnection);
	
	@KSOAP(cacheable=true) public String getVRDEExtPack();
	@Asyncronous public void setVRDEExtPack(@KSOAP("VRDEExtPack") String vrdeExtPack);
	
	@KSOAP(cacheable=true) public String getAuthLibrary();
	@Asyncronous public void setAuthLibrary(@KSOAP("authLibrary") String authLibrary);
	
	@KSOAP(cacheable=true) public String[] getVRDEProperties();
	
	@Asyncronous public void setVRDEProperty(@KSOAP("key") String key, @KSOAP("value") String value);
	
	@KSOAP(cacheable=true) public String getVRDEProperty(@KSOAP("key") String key);
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IVRDEServer.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.api.jaxb.AuthType
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy


@KsoapProxy
@Ksoap
interface IVRDEServer : IManagedObjectRef {
    companion object {
        const val BUNDLE = "vrde"
        const val PROPERTY_PORT = "TCP/Ports"
    }

    @Cacheable("enabled")
	suspend fun getEnabled(): Boolean
    suspend fun setEnabled(@Cacheable("enabled") enabled: Boolean)

    @Cacheable("authType")
	suspend fun getAuthType(): AuthType
    suspend fun setAuthType(@Cacheable("authType") authType: AuthType)

    @Cacheable("authTimeout")
	suspend fun getAuthTimeout(): Int
    suspend fun setAuthTimeout(@Cacheable("authTimeout") @Ksoap(type = "unsignedInt") authTimeout: Int)

    @Cacheable("allowMultiConnection")
	suspend fun getAllowMultiConnection(): Boolean
    suspend fun setAllowMultiConnection(@Cacheable("allowMultiConnection") allowMultiConnection: Boolean)

    @Cacheable("ReuseSingleConnection")
	suspend fun getReuseSingleConnection(): Boolean
    suspend fun setReuseSingleConnection(@Cacheable("reuseSingleConnection") reuseSingleConnection: Boolean)

    @Cacheable("VRDEExtPack")
	suspend fun getVRDEExtPack(): String
    suspend fun setVRDEExtPack(@Cacheable("VRDEExtPack") VRDEExtPack: String)

    @Cacheable("authLibrary")
	suspend fun getAuthLibrary(): String
    suspend fun setAuthLibrary(@Cacheable("authLibrary") authLibrary: String)

    @Cacheable("VRDEProperties")
	suspend fun getVRDEProperties(): Array<String>

    suspend fun setVRDEProperty(key: String, value: String)

    suspend fun getVRDEProperty(key: String): String
}